src/components/spaces/ViewSpace.tsx

```diff
@@ -223,6 +224,13 @@ ViewSpacePage.getInitialProps = async (props): Promise<Props> => {
     return return404(props)
   }
 
+  const handle = `@${spaceData.struct.handle.unwrapOr(undefined)}`
+
+  if (handle !== idOrHandle && res) {
```

Comment:
We should redirect to a space handle only if a space has a handle and the current part of URL is not a handle but id.
```suggestion
  if (handle && handle !== idOrHandle && res) {
```

Reply:
Or compare the current URL with an expected valid URL like you do on ViewPost component:
```
  if (currentPostUrl !== validPostUrl && res) {
```